source/developer-guides/debug.rst

```diff
@@ -0,0 +1,197 @@
+    
+  
+3. Build SBL source code
+
+4. Stitch IFWI and flash SBL Image on system board
+
+|
+
+**Step 2 - Configure Intel UDK Debugger Environment on Host**
+
+
+1. Configuration file default location 
+
+ - Windows:: 
+ 
+    C:\Program Files (x86)\Intel\Intel(R) UEFI Development Kit Debugger Tool\SoftDebugger.ini 
+ 
+ - Linux::
+ 
+    /etc/udkdebugger.conf
+    
+
+2. [Debug Port] option::
+
+    [Debug Port]
+    Channel = Serial    <== Must be Serial
+    Port = COM5         <== Change properly
+    FlowControl = 0     <== 0 for now**
+    BaudRate = 115200   <== Change properly
+    Server =            <== Can be empty
+
+
+    
+  For TCP port configuration::
+  
+    [Debug Port]
+    Channel = TCP
+    Server = 127.0.0.1
+    Port = 1234
+
+|
+
+  For PIPE port configuration::
+  
+    [Debug Port]
+    Channel = Pipe
+    Port = /tmp/serial
+
+|
+.. tip::
+    How to create a pipe in Linux::
+    
+        mkfifo /tmp/serial.in
+        
+        mkfifo /tmp/serial.out
+    
+    
+    
+3. [Target System] option::
+
+    [Target System]
+    FlashRange        = 0xFEF00000:0x1100000     <== This must be added for APL platform for code execution debugging in CAR
+ 
+|    
+    
+  For **QEMU** target specific, set NoAccessLimit option is required::
+    
+    [Target System]
+    FlashRange = 0xFFE00000:0x200000
+    NoAccessLimit = 0
+    
+
+4. [Maintenance] option::
+
+    [Maintenance]
+    Trace=0x10          <== This is optional. 0x0: Disable trace output, 0x3f: Enable full trace output
+
+|
+
+**Step 3 - Launch Intel UDK Debugger on Host**
+
+* Windows
+ - Launch "**Start WinDbg with Intel UDK Debugger Tool**"
+
+
+.. image:: /images/start_windbg.jpg
+   :alt: Compile completed
+   :align: center
+   :width: 640px
+   :height: 480px
+
+
+
+
+
+* Linux
+ - Launch::
+
+    /opt/intel/udkdebugger/bin/udk-gdb-server
+
+
+.. note::
+  To debug QEMU with TCP port configuration, execute **Step 4** before **Step 3**.
+
+|
+
+**Step 4 - Power on TARGET platform**
+
+* Windows
+ - Power on TARGET platfrom, the connection of HOST and TARGET will be established immediately.
+
+* Linux
+ - Power on TARGET platform
+ - Launch GDB in a separate terminal
+ - Make a connection with GDB target command::
+    
+    target remote :1234 
+    
+   (or target remote FULL_SERVER_URL:1234 from Intel UDK Debugger console)
+  
+  
+ - Run Intel UDK Debugger scripts for GDB::
+
+    source /opt/intel/udkdebugger/script/udk_gdb_script
+    
+|
+
+* To turn on **QEMU** target with TCP port configuration, follow run qemu-system first then launch Intel UDK Debugger as Step 3
+
+  * Windows::
+  
+      qemu-system-x86_64.exe -m 256M -machine q35 -nographic -pflash Outputs/qemu/SlimBootloader.bin -serial tcp:127.0.0.1:1234,server
+
+  * Linux::
+  
+      qemu-system-x86_64 -m 256M -machine q35 -nographic -pflash Outputs/qemu/SlimBootloader.bin -serial tcp:127.0.0.1:1234,server
+
+
+|
+
+**Step 5 - Start debugging**
+
+Start debugging with WinDbg or GDB
```

Comment:
Add a tip here: 

Sometimes UDK tool does not terminate itself completely after exit. Look for process name ‘eXDI’ and kill it before starting a new debugging session.